src/Payum/Paypal/Rest/Tests/PaypalRestGatewayFactoryTest.php

```diff
@@ -72,6 +72,38 @@ public function shouldAllowCreateGateway()
         $this->assertAttributeNotEmpty('extensions', $extensions);
     }
 
+    /**
+     * @test
+     */
+    public function shouldAllowCreateGatewayWithCustomConfig()
+    {
+        $factory = new PaypalRestGatewayFactory();
+
+        $givenConfig = [
+            'log.LogLevel' => 'DEBUG',
+            'mode' => 'live',
+            'log.FileName' => '/foo/bar.log',
+            'http.ConnectionTimeOut' => '10',
+        ];
+
+        $gateway = $factory->create([
+            'client_id' => 'cId',
+            'client_secret' => 'cSecret',
+            'config' => $givenConfig,
+        ]);
+
+        $apis = $this->readAttribute($gateway, 'apis');
+        $apiContext = null;
+        foreach ($apis as $api) {
+            if ($api instanceof \PayPal\Rest\ApiContext) {
+                $apiContext = $api;
```

Comment:
Once you have the required API, the loop can be broken

```suggestion
                $apiContext = $api;
                break;
```

Reply:
👍  Good catch :)! Thanks